### PR TITLE
[DATA-693] initial cartographer formula

### PR DESCRIPTION
**Pull request by JohnN193**
brew formula for installing cartographer. Currently tested on M1 mac and x86 mac.

https://viam.atlassian.net/browse/DATA-693

### CODE REVIEW

File: Formula/carto-grpc-server.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+class CartoGrpcServer < Formula
+  desc "A Viam slam GRPC server for Cartographer"
+  homepage "https://www.viam.com/"
+  url "https://github.com/viamrobotics/slam.git",
+    tag: "v0.1.10"
```
Comment (JohnN193) [inline Formula/carto-grpc-server.rb:5]: will not merge until next release occurs in order to include some necessary commits from the slam repo

**Review by Otterverse — CHANGES_REQUESTED** [review]
So it looks mostly good (save a nit below about formatting) But... cartographer itself fails to compile under linux, both against 0.1.10 and with --HEAD.

The error I hit is this:

```
/tmp/carto-grpc-server-20221208-1460499-kyx5mh/slam-libraries/viam-cartographer/src/io/file_handler.cc: In function 'double viam::io::ReadTimeFromTimestamp(std::string)':
/tmp/carto-grpc-server-20221208-1460499-kyx5mh/slam-libraries/viam-cartographer/src/io/file_handler.cc:104:16: error: 'get_time' is not a member of 'std'; did you mean 'gmtime'?
  104 |     ss >> std::get_time(&dt, time_format.c_str());
      |                ^~~~~~~~
      |                gmtime
```
I'm pretty sure that's just because the file in question needs `#include <iomanip>` (which is where get_time now resides on many compilers)

That needs to be resolved before this goes into a public brew tap.

File: Formula/carto-grpc-server.rb (hunk 1 above)
Comment (tessavitabile) [inline Formula/carto-grpc-server.rb:5]: I found for orbslam that it was necessary to specify the revision along with the tag in order for submodules to be downloaded when installing without `--HEAD`.

**Review by tessavitabile — CHANGES_REQUESTED** [review]
Looks good!

File: Formula/carto-grpc-server.rb (hunk 1 above)
Comment (tessavitabile) [reply Formula/carto-grpc-server.rb:5]: Huh, what I found for orbslam was that when installing without `--HEAD`, it downloads the artifacts associated with the tag, which don't include submodules.

**Comment by JohnN193** [comment]
Kk updated the slam repo to include the missing include, just need an updated version tag for slam

**Review by Otterverse — APPROVED** [review]
LGTM, provided you wait for an updated tag that builds without "--HEAD"

**Review by tessavitabile — APPROVED** [review]
LGTM. It's fine to do a patch release for SLAM yourself. Let me know if you need any information on how to do that.

**Pull request merged**